lte/gateway/c/session_manager/test/test_upf_node_state.cpp

```diff
@@ -90,6 +95,48 @@ TEST_F(SetUPFNodeState, test_upf_node_state) {
   EXPECT_TRUE(upf_n3_addr == ipv4_addr);
 }
 
+// Testing the functionality of SetUPFSessionConfig
+TEST_F(SetUPFNodeState, test_upf_session_config) {
+  // Validate SetUPFSessionConfig
+  ON_CALL(*set_interface_for_up_mock, SetUPFSessionConfig(_, _, _))
+      .WillByDefault(Return(Status::OK));
+
+  auto& ses_config = *sess_config;
+  int32_t count    = 0;
+    for (int i = 0; i < ses_config.upf_session_state_size(); i++) {
+      // Deleting the IMSI prefix from imsi
+      std::string imsi_upf = ses_config.upf_session_state(i).subscriber_id();
+      std::string imsi     = imsi_upf.substr(4, imsi_upf.length() - 4);
+      uint32_t version     = ses_config.upf_session_state(i).session_version();
+      uint32_t teid        = ses_config.upf_session_state(i).local_f_teid();
+      auto session_map     = session_store->read_sessions({imsi});
+      /* Search with session search criteria of IMSI and session_id and
+       * find  respective sesion to operate
+       */
+      SessionSearchCriteria criteria(imsi, IMSI_AND_TEID, teid);
+      auto session_it = session_store->find_session(session_map, criteria);
+      if (!session_it) {
+        EXPECT_FALSE(session_it);
```

Comment:
What does this line do? If the condition is `!session_it` isn't this always going to be false?

Reply:
session_it is not always false, sometimes If we don't find a session with SessionSearchCriteria() continue with next imsi + SearchCriteria. So, I wanted to validate the false case here.

Reply:
session_it is not always false. Sometimes if we don't find session with SessionSearchCriteria(), we will continue with next imsi+SearchCriteria , So I wanted to validate the false case here.